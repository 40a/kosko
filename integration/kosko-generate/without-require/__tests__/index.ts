import execa from "execa";
import { dirname } from "path";
import { runCLI } from "../../../run";

let args: string[];
let result: execa.ExecaReturns;

beforeEach(async () => {
  result = await runCLI(args, {
    cwd: dirname(__dirname)
  });
});

describe("when output is not set", () => {
  beforeAll(() => {
    args = ["generate", "--env", "dev"];
  });

  test("should output YAML", () => {
    expect(result.stdout).toMatchSnapshot();
  });
});

describe("when output = json", () => {
  beforeAll(() => {
    args = ["generate", "--env", "dev", "--output", "json"];
  });

  test("should output JSON", () => {
    expect(result.stdout).toMatchSnapshot();
  });
});
