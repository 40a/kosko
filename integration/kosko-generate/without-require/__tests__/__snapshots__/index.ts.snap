// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when output = json should output JSON 1`] = `
"{
  \\"apiVersion\\": \\"v1\\",
  \\"kind\\": \\"List\\",
  \\"items\\": [
    {
      \\"apiVersion\\": \\"apps/v1\\",
      \\"kind\\": \\"Deployment\\",
      \\"metadata\\": {
        \\"name\\": \\"nginx\\"
      },
      \\"spec\\": {
        \\"replicas\\": 1,
        \\"selector\\": {
          \\"matchLabels\\": {
            \\"app\\": \\"nginx\\"
          }
        },
        \\"template\\": {
          \\"metadata\\": {
            \\"labels\\": {
              \\"app\\": \\"nginx\\"
            }
          },
          \\"spec\\": {
            \\"containers\\": [
              {
                \\"name\\": \\"nginx\\",
                \\"image\\": \\"nginx\\"
              }
            ]
          }
        }
      }
    },
    {
      \\"apiVersion\\": \\"v1\\",
      \\"kind\\": \\"Service\\",
      \\"metadata\\": {
        \\"name\\": \\"nginx\\"
      },
      \\"spec\\": {
        \\"selector\\": {
          \\"app\\": \\"nginx\\"
        },
        \\"ports\\": [
          {
            \\"port\\": 80
          }
        ]
      }
    }
  ]
}"
`;

exports[`when output = yaml should output YAML 1`] = `
"---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx
spec:
  replicas: 1
  selector:
    matchLabels:
      app: nginx
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
        - name: nginx
          image: nginx
---
apiVersion: v1
kind: Service
metadata:
  name: nginx
spec:
  selector:
    app: nginx
  ports:
    - port: 80"
`;

exports[`when output is invalid should print the error 1`] = `
"kosko generate [components..]

Generate Kubernetes manifests

Positionals:
  components  Components to generate. This overrides components set in config
              file.                                        [array] [default: []]

Global Options:
  --cwd      Path of working directory                   [string] [default: CWD]
  --silent   Disable log output                       [boolean] [default: false]
  --help     Show help                                                 [boolean]
  --version  Show version number                                       [boolean]

Options:
  --env, -e      Name of environment                                    [string]
  --output, -o   Output format
                            [string] [choices: \\"yaml\\", \\"json\\"] [default: \\"yaml\\"]
  --require, -r  Require modules. Modules set in config file will also be
                 required.                                 [array] [default: []]

Examples:
  kosko generate                      Generate components
  kosko generate foo bar              Specify components
  kosko generate foo_*                Use glob syntax
  kosko generate --env foo            Set environment
  kosko generate -r ts-node/register  Require external modules

Invalid values:
  Argument: output, Given: \\"foo\\", Choices: \\"yaml\\", \\"json\\"
"
`;

exports[`when output is not set should output YAML 1`] = `
"---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx
spec:
  replicas: 1
  selector:
    matchLabels:
      app: nginx
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
        - name: nginx
          image: nginx
---
apiVersion: v1
kind: Service
metadata:
  name: nginx
spec:
  selector:
    app: nginx
  ports:
    - port: 80"
`;
